### PR TITLE
Bug 1299065: Fix etcd hostname/ip address handling

### PR DESCRIPTION
**Pull request by abutcher**
* Deprecate use of `etcd_interface` inventory variable.
* Utilize `etcd_hostname` and `etcd_ip` variables for etcd configuration and certificates. Values for these variables are defaulted in the `etcd_common` role.
* Adds `openshift_etcd_facts` role which is currently only responsible for overriding `etcd_hostname` and `etcd_ip` (values are also set in `playbooks/common/openshift-etcd/config.yml`)
* Adds `openshift_etcd_certificates` meta role for ensuring vars are set for `etcd_certificates` and `etcd_ca`.

Etcd certificates and configuration are based on the `openshift_hostname` and `openshift_ip` set in inventory, for example:

```
[etcd]
etcd1.example.com openshift_hostname=etcd1.example.com openshift_ip=192.168.122.220
etcd2.example.com openshift_hostname=etcd2.example.com openshift_ip=192.168.122.221
etcd3.example.com openshift_hostname=etcd3.example.com openshift_ip=192.168.122.222
```

### CODE REVIEW

File: roles/etcd/defaults/main.yaml [hunk 1]
```diff
@@ -9,10 +8,10 @@ etcd_peer_url_scheme: http
 etcd_initial_cluster_state: new
 etcd_initial_cluster_token: etcd-cluster-1
 
-etcd_initial_advertise_peer_urls: "{{ etcd_peer_url_scheme }}://{{ hostvars[inventory_hostname]['ansible_' + etcd_interface]['ipv4']['address'] }}:{{ etcd_peer_port }}"
-etcd_listen_peer_urls: "{{ etcd_peer_url_scheme }}://{{ hostvars[inventory_hostname]['ansible_' + etcd_interface]['ipv4']['address'] }}:{{ etcd_peer_port }}"
-etcd_advertise_client_urls: "{{ etcd_url_scheme }}://{{ hostvars[inventory_hostname]['ansible_' + etcd_interface]['ipv4']['address'] }}:{{ etcd_client_port }}"
-etcd_listen_client_urls: "{{ etcd_url_scheme }}://{{ hostvars[inventory_hostname]['ansible_' + etcd_interface]['ipv4']['address'] }}:{{ etcd_client_port }}"
+etcd_initial_advertise_peer_urls: "{{ etcd_peer_url_scheme }}://{{ hostvars[inventory_hostname].openshift.common.ip }}:{{ etcd_peer_port }}"
```
Comment (detiber) [inline roles/etcd/defaults/main.yaml]: We should avoid using openshift.common.ip in this role, rather we should map it to an 'etcd_' variable in a new openshift_etcd_facts role

Comment (abutcher) [inline roles/etcd/defaults/main.yaml]: I can shuffle that work to this PR, that's what I'm doing in the delegate PR.

Comment (abutcher) [inline roles/etcd/defaults/main.yaml]: I've put the hybrid approach together for this PR just incase the delegate PR slips (good chance it will).

**Comment by detiber** [comment]
aos-ci-test

**Comment by detiber** [comment]
lgtm pending tests

**Comment by abutcher** [comment]
@detiber Updated to remove `etcd_interface` variable and to remove the `openshift_docker` dep from the `etcd` role (`openshift_etcd` maintains the dep)

**Comment by abutcher** [comment]
aos-ci-test

File: roles/etcd/meta/main.yml [hunk 2]
```diff
@@ -16,6 +16,5 @@ galaxy_info:
   - cloud
   - system
 dependencies:
-- { role: openshift_docker }
```
Comment (abutcher) [inline roles/etcd/meta/main.yml]: @brenton Heads up! `openshift_etcd` depends on `openshift_docker` conditionally.

Comment (brenton) [inline roles/etcd/meta/main.yml]: Thanks for doing this.  Have you added that commit yet?

Comment (abutcher) [inline roles/etcd/meta/main.yml]: That dependency was already there so we should probably double check things with this removed.

**Comment by abutcher** [comment]
aos-ci-test

**Comment by detiber** [comment]
:+1: on the latest changes

**Comment by detiber** [comment]
@brenton

**Comment by detiber** [comment]
@abutcher found a regression with this, etcd_hostname and etcd_ip is not currently being set for the hosts in playbooks/common/openshift-master/config.yml before calling openshfit_etcd_certificates.

**Comment by abutcher** [comment]
@detiber zoinks! https://github.com/openshift/openshift-ansible/pull/1785

**Pull request merged**